src/Microsoft.IdentityModel.Tokens/ECDsaAdapter.cs

```diff
@@ -203,7 +203,7 @@ private static uint GetKeyByteCount(string curveId)
                     keyByteCount = 66;
                     break;
                 default:
-                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10645, curveId)));
+                    throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX10645, LogHelper.MarkAsNonPII(curveId))));
```

Comment:
@cesarpayan I know you're looking into ECD stuff, is the curve ID secret or is that like the public key. I know very little about this stuff.

Reply:
The curve ID is part of the public key. See [RFC 7518 §6.2.1 Parameters for Elliptic Curve Public Keys](https://www.rfc-editor.org/rfc/rfc7518#section-6.2.1) and [IANA registry for JSON Web Key Elliptic Curve](https://www.iana.org/assignments/jose/jose.xhtml#web-key-elliptic-curve).

Reply:
If the call stack is as follows, then the public key belongs to the identity provider and not to the individual user.

- [Microsoft.IdentityModel.Protocols.OpenIdConnect.OpenIdConnectConfigurationRetriever.GetAsync](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfigurationRetriever.cs#L100>)
- [Microsoft.IdentityModel.Tokens.JsonWebKeySet.GetSigningKeys](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/JsonWebKeySet.cs#L165>)
- [Microsoft.IdentityModel.Tokens.JsonWebKeyConverter.TryConvertToECDsaSecurityKey](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/JsonWebKeyConverter.cs#L353>)
- [Microsoft.IdentityModel.Tokens.ECDsaSecurityKey(JsonWebKey webKey, bool usePrivateKey)](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/ECDsaSecurityKey.cs#L44>)
- [Microsoft.IdentityModel.Tokens.ECDsaAdapter.CreateECDsa](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/ECDsaAdapter.cs#L75>)
- [Microsoft.IdentityModel.Tokens.ECDsaAdapter.CreateECDsaUsingCNGKey](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/ECDsaAdapter.cs#L102>)
- [Microsoft.IdentityModel.Tokens.ECDsaAdapter.GetKeyByteCount](<https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/fc59ad574ee0e0b9026c405fb566f343bc314bd7/src/Microsoft.IdentityModel.Tokens/ECDsaAdapter.cs#L206>)

The code in this repo doesn't seem to ever call `internal ECDsaSecurityKey(JsonWebKey webKey, bool usePrivateKey)` with `usePrivateKey: true`, not even in tests. Applications cannot do that either, because the constructor is `internal`.

Reply:
Thank you for the explanation @KalleOlaviNiemitalo